Samples/DynamicDependencies/DirectX/direct3d9ex/BackgroundThread.cpp

```diff
@@ -0,0 +1,380 @@
+//
+// Copyright(c) Microsoft Corporation.
+// Licensed under the MIT License.
+//--------------------------------------------------------------------------------------
+#include <windows.h>
+#include <d3d9.h>
+#include <d3dx9.h>
+
+#define SAFE_RELEASE(p) { if(p) { (p)->Release(); (p) = NULL; } }
+
+//-----------------------------------------------------------------------------
+// structures
+//-----------------------------------------------------------------------------
+struct BACKGROUNDVERTEX
+{
+    FLOAT x, y, z;	  // The untransformed position for the vertex
+    DWORD dwColor;	  // Color
+};
+#define D3DFVF_BACKGROUNDVERTEX (D3DFVF_XYZ|D3DFVF_DIFFUSE)
+
+//--------------------------------------------------------------------------------------
+// Global variables
+//--------------------------------------------------------------------------------------
+bool g_bEndThread = false;
+IDirect3DTexture9*	g_pRenderTarget = NULL;
+IDirect3DTexture9*	g_pSharedTexture = NULL;
+IDirect3DVertexBuffer9* g_pCubeVB = NULL;
+IDirect3DIndexBuffer9* g_pCubeIB = NULL;
+HANDLE g_SharedHandle = NULL;
+UINT g_RTWidth = 1024;
+UINT g_RTHeight = 1024;
+int g_CubeCubes = 3;
+float g_BoxRad = 30.0f;
+CRITICAL_SECTION g_CSCubes;
+
+float g_fBackLastFrameTime = 0.0f;
+LARGE_INTEGER g_liBackLastTimerUpdate = {0};
+LARGE_INTEGER g_liBackTimerFrequency = {0};
+
+//--------------------------------------------------------------------------------------
+// Function Prototypes and Externs
+//--------------------------------------------------------------------------------------
+extern HRESULT CreateD3D9VDevice( IDirect3D9Ex* pD3D, IDirect3DDevice9Ex** ppDev9Ex, D3DPRESENT_PARAMETERS* pD3DPresentParameters, HWND hWnd );
+DWORD WINAPI BackgroundThreadProc( LPVOID lpParam );
+HRESULT CreateCube(IDirect3DDevice9Ex* pDev);
+HRESULT CreateSharedRenderTexture(IDirect3DDevice9Ex* pDev);
+void RenderBackground(IDirect3DDevice9Ex* pDev);
+void CleanupBackground();
+
+
+//--------------------------------------------------------------------------------------
+// Exported
+//--------------------------------------------------------------------------------------
+HANDLE CreateBackgroundThread(IDirect3D9Ex* pD3D9);
+void KillBackgroundThread() { g_bEndThread = true; }
+HANDLE GetSharedTextureHandle() { return g_SharedHandle; }
+int IncreaseCubeCount();
+int DecreaseCubeCount();
+float GetFPS();
+
+
+// main background thread proc
+DWORD WINAPI BackgroundThreadProc( LPVOID lpParam )
+{
+	// Create a critsec
+	InitializeCriticalSection( &g_CSCubes );
+
+	// Create a d3d9V device
+	IDirect3DDevice9Ex*		pDevBackground = NULL;
+	IDirect3D9Ex* pD3D9 = (IDirect3D9Ex*)(lpParam);
+
+    D3DPRESENT_PARAMETERS d3dpp = {0};
+
+    d3dpp.Windowed = TRUE;
+    d3dpp.SwapEffect = D3DSWAPEFFECT_DISCARD;
+    d3dpp.BackBufferFormat = D3DFMT_UNKNOWN;
+    d3dpp.PresentationInterval = D3DPRESENT_INTERVAL_DEFAULT;
+
+	if( FAILED( CreateD3D9VDevice( pD3D9, &pDevBackground, &d3dpp, NULL ) ) )
```

Comment:
@chrisguzak @oldnewthing is there a standard for spacing in these cpp files? For example:

```cpp
if( FAILED( CreateD3D9VDevice( pD3D9, &pDevBackground, &d3dpp, NULL ) ) )

if  FAILED(CreateD3D9VDevice(pD3D9, &pDevBackground, &d3dpp, NULL)))
```

and same for method calls?

Reply:
this is an existing sample from the Win7 era, I would not want to fixup all the old whitespace.